### PR TITLE
test: try to make reset-modal.spec reliable

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
- **test: clear before typing + improve readability**
- **test: increase wait in case preview is slow**

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

This is mostly a refactor, but there are two real changes: 1) clear the editor before typing 2) wait longer for the preview to first appear.

### CODE REVIEW

File: e2e/reset-modal.spec.ts [hunk 1]
```diff
@@ -44,21 +44,34 @@ test('should render the modal content correctly', async ({ page }) => {
...
-  await expect(
-    page
-      .frameLocator('iframe[title="challenge preview"]')
-      .getByRole('heading', { name: 'Cat Photos' })
-  ).toBeVisible();
+  // Modify the text in the editor pane, clearing first, otherwise the existing
+  // text will be selected before typing
+  await page.getByLabel(editorPaneLabel).fill('');
+  await page.getByLabel(editorPaneLabel).fill(updatedText);
+  await expect(updatedFrame).toBeVisible();
```
Comment (naomi-lgbt) [inline e2e/reset-modal.spec.ts:71]:
```suggestion
  await expect(updatedFrame).toBeVisible({ timeout: 10000});
```

We should maybe put a wait in here too - sometimes the preview reload can take a minute.

File: e2e/reset-modal.spec.ts [hunk 2]
```diff
@@ -69,16 +82,14 @@ test('User can reset challenge', async ({ page }) => {
...
     .getByRole('button', { name: translations.buttons['reset-lesson'] })
     .click();
 
-  await expect(
-    page
-      .frameLocator('iframe[title="challenge preview"]')
-      .getByRole('heading', { name: 'Cat Photos' })
-  ).not.toBeVisible();
+  // Check it's back to the initial state
+  await expect(initialFrame).toBeVisible();
```
Comment (naomi-lgbt) [inline e2e/reset-modal.spec.ts:92]:
```suggestion
  await expect(initialFrame).toBeVisible({ timeout: 10000 });
```

File: e2e/reset-modal.spec.ts (hunk 1 above)
Comment (ojeytonwilliams) [reply e2e/reset-modal.spec.ts:71]: Can't hurt.

**Review by naomi-lgbt — APPROVED** [review]

**Comment by ojeytonwilliams** [comment]
Zero flake across all 3 browsers :smiling_face_with_tear:

**Comment by naomi-lgbt** [comment]
I wonder if we should do this for that one syntax error test in the js challenges

**Comment by ojeytonwilliams** [comment]
https://github.com/freeCodeCamp/freeCodeCamp/pull/54505 should have fixed it.

**Review by moT01 — APPROVED** [review]

**Pull request merged**